bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaGatewayHandler.java

```diff
@@ -52,21 +65,47 @@ public void initializeThing(@Nullable ThingStatus bridgeStatus) {
 
     @Override
     public void refresh(String channel) {
-        String id = getGateWayId();
-        SomfyTahomaStatus status = getTahomaStatus(id);
-        String tahomaStatus = status.getStatus();
-        Channel ch = thing.getChannel(channel);
-        if (ch != null) {
-            updateState(ch.getUID(), new StringType(tahomaStatus));
-        }
-        // update the firmware property
-        String fw = status.getProtocolVersion();
-        updateProperty(PROPERTY_FIRMWARE_VERSION, fw);
+        if (channel.equals(STATUS)) {
+            String id = getGateWayId();
+            SomfyTahomaStatus status = getTahomaStatus(id);
+            String tahomaStatus = status.getStatus();
+            Channel ch = thing.getChannel(channel);
+            if (ch != null) {
+                updateState(ch.getUID(), new StringType(tahomaStatus));
+            }
+            // update the firmware property
+            String fw = status.getProtocolVersion();
+            updateProperty(PROPERTY_FIRMWARE_VERSION, fw);
 
-        updateStatus("DISCONNECTED".equals(tahomaStatus) ? ThingStatus.OFFLINE : ThingStatus.ONLINE);
+            updateStatus("DISCONNECTED".equals(tahomaStatus) ? ThingStatus.OFFLINE : ThingStatus.ONLINE);
+        } else if (channel.equals(SCENARIOS)) {
+            SomfyTahomaBridgeHandler handler = getBridgeHandler();
+            if (handler != null) {
+                List<StateOption> options = new ArrayList<>();
+                for (SomfyTahomaActionGroup actionGroup : handler.listActionGroups()) {
+                    options.add(new StateOption(actionGroup.getOid(), actionGroup.getLabel()));
+                }
+                stateDescriptionProvider.setStateOptions(new ChannelUID(getThing().getUID(), channel), options);
+            }
+        }
     }
 
     public String getGateWayId() {
         return getThing().getConfiguration().get("id").toString();
     }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        super.handleCommand(channelUID, command);
+        if (command instanceof RefreshType) {
+            return;
+        }
+        if (channelUID.getId().equals(SCENARIOS)) {
+            SomfyTahomaBridgeHandler handler = getBridgeHandler();
+            if (handler != null && command instanceof StringType) {
+                handler.executeActionGroup(command.toString());
+                updateState(channelUID, new StringType(command.toString()));
```

Comment:
What is the reason for setting the Channel with the updated value manually?

Reply:
Because the channel state was not set if I did not add this line.
I tried only with MainUI. Could it be a setup problem with my item or a UI bug ?
I will try again, including with Basic UI.

Reply:
Did you try to set the auto update policy to `recommended`? https://www.openhab.org/docs/developer/bindings/thing-xml.html#auto-update-policies

Reply:
Why the default policy would not work?

Reply:
I'm not aware of the details of the auto update algorithm, but I think the default setting expects the Channel to be updated when the Thing is polled the next time. This will never happen in your case, so you can adjust the auto update policy to tell the framework to update the Channel immediately. Then, you can remove the `updateState()` and the model will also tell one the intented behavior of the Channel.

Reply:
I agree that it is the good approach. Tested and working.